master/buildbot/reporters/base.py

```diff
@@ -57,10 +57,6 @@ def reconfigService(self, generators):
             yield consumer.stopConsuming()
         self._event_consumers = []
 
-        if self._pending_got_event_call is not None:
```

Comment:
The reason for this wait is that the callback code may depend on the exact configuration, so we should wait until it completes before changing the configuration under the callback's feet.

Unfortunately even this code did not prevent race conditions because the deriving class will run its `reconfigService` only after `startConsuming()`, so new `_got_event` calls could happen before reconfiguration is finished. Thus I think it's OK to remove it and later we will need a proper solution.

Reply:
I filed #6981 and #6982 so that these problems are not forgotten.